conman/routes/models.py

```diff
@@ -74,6 +84,64 @@ def handle(self, request, path):
         # Deal with the request
         return handler.handle(request, path)
 
+    def move_to(self, new_url, *, move_children):
+        """
+        Move this Route to a new url.
+
+        Requires `move_children` as a keyword argument. If True, all child
+        Routes will be moved along with this parent. Otherwise, when False, any
+        and all children will remain where they are.
+
+        In the event of a clash, an IntegrityError will be raised.
+        """
+        if move_children:
+            Route.objects.move_branch(self.url, new_url)
+            self.url = new_url
+        else:
+            old_url, self.url = self.url, new_url
+            try:
+                self.save()
+            except IntegrityError:
+                self.url = old_url
+                raise
```

Comment:
Come to think of it, given this method only ever executes one query, you could argue that it's already atomic. The only difference is that of not automatically allowing further queries after the  `IntegrityError`, which may not always be required.

I'll add `atomic` to `swap_with`, as that does multiple queries, but unless you think it's important that this changes, I'll leave it as is.